File: complainator_ui/src/dto/SuggestionDto.ts
export type SuggestionStatus = "Pending" | "Accepted" | "Rejected";

export interface GenerateSuggestionsResponse {
  suggestions: SuggestionDto[];
}

export interface SuggestionDto {
  id: string; // Guid -> string
  suggestionText: string;
  status: SuggestionStatus;
}

export interface UpdateSuggestionRequest {
  status: SuggestionStatus;
}

File: ComplainatorAPI/DTO/SuggestionDto.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using ComplainatorAPI.Domain.Entities;

namespace ComplainatorAPI.DTO
{
    public class GenerateSuggestionsResponse
    {
        public List<SuggestionDto> Suggestions { get; set; } = new();
    }

    public class SuggestionDto
    {
        public Guid Id { get; set; }
        public string SuggestionText { get; set; } = string.Empty;
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public SuggestionStatus Status { get; set; }
    }

    public class UpdateSuggestionRequest
    {
        [Required]
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public SuggestionStatus Status { get; set; }
    }
} 